File: src/code/fastpow/fastpow.js
"use strict";
function fastPow(a, n) {
    const $ = BigInt;
    const MOD = $(1e9 + 7);
    let b = $(a) % MOD, res = 1n;
    while (n) {
        if (n % 2 === 1) {
            res = res * b % MOD;
        }
        b = b * b % MOD;
        n = Math.trunc(n / 2);
    }
    return Number(res % MOD);
}

File: src/code/fastpow/fast_pow.go
func fastPow(a, n, m int64) int64 {
    b := a % m
    res := int64(1)
    for n > 0 {
        if n & 1 == 1 {
            res = (res * b) % m
        }
        b = b * b % m
        n = n / 2
    }
    return res % m
}